symplyphysics/laws/dynamics/vector/acceleration_from_force.py

```diff
@@ -41,24 +41,26 @@ def force_law(acceleration_: Vector) -> Vector:
 # Condition: mass is constant
 
 time = force_momentum_law.time
-momentum_x = Function("momentum_x", units.mass * units.velocity)
-momentum_y = Function("momentum_y", units.mass * units.velocity)
-momentum_z = Function("momentum_z", units.mass * units.velocity)
 
+momentum_x = Function("momentum_x", units.momentum)
+momentum_y = Function("momentum_y", units.momentum)
+momentum_z = Function("momentum_z", units.momentum)
 momentum_vec = Vector([momentum_x(time), momentum_y(time), momentum_z(time)])
+
 force_derived = force_momentum_law.force_law(momentum_vec)
 
 momentum_def_sub = momentum_def.definition.subs(momentum_def.mass, mass)
-velocity_x = solve(momentum_def_sub, momentum_def.velocity)[0].subs(momentum_def.momentum, momentum_x(time))
-velocity_y = solve(momentum_def_sub, momentum_def.velocity)[0].subs(momentum_def.momentum, momentum_y(time))
-velocity_z = solve(momentum_def_sub, momentum_def.velocity)[0].subs(momentum_def.momentum, momentum_z(time))
+velocity_vec = Vector([
+    solve(momentum_def_sub, momentum_def.velocity)[0].subs(momentum_def.momentum, momentum_component)
```

Comment:
Solve it once, outside of this list generator.

Reply:
Done